bin/bl-exit

```diff
@@ -1,154 +1,425 @@
+            self.openbox_exit()
+            gtk.main_quit()
+
+        def suspend_action(self, button):
+            self.disable_buttons()
+            self.selected_action = 'suspend'
+            self.status.set_label("Suspending, please standby...")
+            self.do_action("Suspend")
+            gtk.main_quit()
+
+        def hibernate_action(self, button):
+            self.disable_buttons()
+            self.selected_action = 'hibernate'
+            self.status.set_label("Hibernating, please standby...")
+            self.do_action("Hibernate")
+            gtk.main_quit()
+
+        def hybridsleep_action(self, button):
+            self.disable_buttons()
+            self.selected_action = 'hybridsleep'
+            self.status.set_label("Hibernating + Sleeping, please standby...")
+            self.do_action("HybridSleep")
+            gtk.main_quit()
+
+        def reboot_action(self, button):
+            self.disable_buttons()
+            self.selected_action = 'reboot'
+            self.status.set_label("Rebooting, please standby...")
+            self.do_action("Reboot")
+            gtk.main_quit()
+
+        def shutdown_action(self, button):
+            self.disable_buttons()
+            self.selected_action = 'poweroff'
+            self.status.set_label("Shutting down, please standby...")
+            self.do_action("PowerOff")
+            gtk.main_quit()
+
 
 def main(argv):
-    try:
-        opts, args = getopt.getopt(argv, "hlsrp", ["help","logout","suspend","reboot","poweroff"])
-    except getopt.GetoptError:
-        print_usage(1)
-    for opt, arg in opts:
-        if opt in ("-h", "--help"):
-            print_usage(0)
-        elif opt in ("-l", "--logout"):
-            logout()
-            sys.exit(0)
-        elif opt in ("-s", "--suspend"):
-            suspend()
-            sys.exit(0)
-        elif opt in ("-r", "--reboot"):
-            reboot()
-            sys.exit(0)
-        elif opt in ("-p", "--poweroff"):
-            poweroff()
-            sys.exit(0)
-
-    Blexit().main()
+    '''
+    The script works both in a graphical and a non-graphical environment.
+
+    In a graphical environment, the BlExitWindow instance is only shown when
+    the script is launched without arguments. The user selects the action she
+    wants by clicking the right button.
+
+    WHen  the script is launched In a graphical environment the requested
+    action should be one of the accepted arguments and the action is executed
+    without asking for confirmation - as if the script was launched from the
+    command line.
+
+    In a non-graphical environment, one of the accepted actions must be
+    specified as an argument.
+    '''
+    if display and len(argv) == 0:
+        blexit = BlexitWindow(argv)
+    else:
+        blexit = BlexitBase(argv)
+    blexit.main()
 
 if __name__ == "__main__":
+    blexit = None
     main(sys.argv[1:])
```

Comment:
@2ion,
Really appreciate your remarks.
On first sight I will be applying all of them.